src/main/java/teammates/storage/sqlentity/Instructor.java

```diff
@@ -0,0 +1,120 @@
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.Table;
+
+/**
+ * Represents an Instructor entity.
+ */
+@Entity
+@Table(name = "Instructors")
+public class Instructor extends User {
+    @Column(nullable = false)
+    private String registrationKey;
+
+    @Column(nullable = false)
+    private boolean isDisplayedToStudents;
+
+    @Column(nullable = false)
+    private String displayName;
+
+    @Column(nullable = false)
+    @Enumerated(EnumType.STRING)
+    private InstructorPermissionRole role;
+
+    @Column(nullable = false)
+    private String instructorPrivileges;
+
+    protected Instructor() {
+        // required by Hibernate
+    }
+
+    public String getRegistrationKey() {
+        return registrationKey;
+    }
+
+    public void setRegistrationKey(String registrationKey) {
+        this.registrationKey = registrationKey;
+    }
+
+    public boolean isDisplayedToStudents() {
+        return isDisplayedToStudents;
+    }
+
+    public void setDisplayedToStudents(boolean displayedToStudents) {
+        isDisplayedToStudents = displayedToStudents;
+    }
+
+    public String getDisplayName() {
+        return displayName;
+    }
+
+    public void setDisplayName(String displayName) {
+        this.displayName = displayName;
+    }
+
+    public InstructorPermissionRole getRole() {
+        return role;
+    }
+
+    public void setRole(InstructorPermissionRole role) {
+        this.role = role;
+    }
+
+    public String getInstructorPrivileges() {
+        return instructorPrivileges;
+    }
+
+    public void setInstructorPrivileges(String instructorPrivileges) {
+        this.instructorPrivileges = instructorPrivileges;
+    }
+
+    @Override
+    public String toString() {
+        return "Instructor [id=" + super.getId() + ", registrationKey=" + registrationKey
+                + ", isDisplayedToStudents=" + isDisplayedToStudents + ", displayName=" + displayName
+                + ", role=" + role + ", instructorPrivileges=" + instructorPrivileges
+                + ", createdAt=" + super.getCreatedAt() + ", updatedAt=" + super.getUpdatedAt() + "]";
+    }
+
+    @Override
+    public int hashCode() {
+        // Instructor Id uniquely identifies an Instructor
+        return super.getId();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        } else if (this == obj) {
+            return true;
+        } else if (this.getClass() == obj.getClass()) {
+            Instructor otherInstructor = (Instructor) obj;
+
+            return Objects.equals(super.getId(), otherInstructor.getId());
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    public void sanitizeForSaving() {
+        // TODO Auto-generated method stub
+    }
+
+    @Override
+    public List<String> getInvalidityInfo() {
+        // TODO Auto-generated method stub
+        return new ArrayList<>();
+    }
```

Comment:
These two methods can be found in `InstructorAttributes` class. Please port them over as well.